phoenix-core/src/main/java/org/apache/phoenix/iterate/MergeSortTopNResultIterator.java

```diff
@@ -79,9 +86,11 @@ public Tuple peek() throws SQLException {
 
     @Override
     public Tuple next() throws SQLException {
-        if (limit >= 0 && count++ >= limit) {
-            return null;
+        while (count < offset) {
+            if (super.next() == null) { return null; }
+            count++;
         }
+        if (limit >= 0 && count++ >= limit) { return null; }
```

Comment:
Actually it should count++ only, but there is a bug in offset count .. which I have updated now.